src/main/java/com/duckblade/osrs/toa/features/nexus/PathLevelTracker.java

```diff
@@ -0,0 +1,118 @@
+package com.duckblade.osrs.toa.features.nexus;
+
+import com.duckblade.osrs.toa.TombsOfAmascutConfig;
+import com.duckblade.osrs.toa.module.PluginLifecycleComponent;
+import com.duckblade.osrs.toa.util.RaidRoom;
+import com.duckblade.osrs.toa.util.RaidState;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import net.runelite.api.Client;
+import net.runelite.api.Varbits;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.api.events.WidgetLoaded;
+import net.runelite.api.widgets.InterfaceID;
+import net.runelite.api.widgets.Widget;
+import net.runelite.client.eventbus.EventBus;
+import net.runelite.client.eventbus.Subscribe;
+
+@Singleton
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+public class PathLevelTracker implements PluginLifecycleComponent
+{
+	private static final int CHILD_ID_KEPHRI = 49;
+	private static final int CHILD_ID_AKKHA = 51;
+	private static final int CHILD_ID_BABA = 53;
+	private static final int CHILD_ID_ZEBAK = 55;
+
+	private final EventBus eventBus;
+	private final Client client;
+
+	@Getter
+	private int kephriPathLevel;
+	@Getter
+	private int akkhaPathLevel;
+	@Getter
+	private int babaPathLevel;
+	@Getter
+	private int zebakPathLevel;
+
+	@Getter
+	private int raidLevel;
+
+	/*
+	 * "For every 5 raid levels, enemy hitpoints, defence, accuracy, and damage increases by 2% additively,
+	 *  with damage capping out at +150%."
+	 * https://oldschool.runescape.wiki/w/Tombs_of_Amascut/Strategies#Invocations_and_Raid_Level
+	 */
+	@Getter
+	private int raidLevelStatMod;
+
+	@Override
+	public boolean isEnabled(final TombsOfAmascutConfig config, final RaidState raidState)
+	{
+		return raidState.getCurrentRoom() == RaidRoom.NEXUS;
+	}
+
+	@Override
+	public void startUp()
+	{
+		eventBus.register(this);
+	}
+
+	@Override
+	public void shutDown()
+	{
+		eventBus.unregister(this);
+	}
+
+	@Subscribe
+	public void onVarbitChanged(final VarbitChanged event)
+	{
+		if (event.getVarbitId() == Varbits.TOA_RAID_LEVEL)
+		{
+			raidLevel = event.getValue();
+			raidLevelStatMod = (raidLevel / 5) * 2;
+		}
+	}
+
+	@Subscribe
+	public void onWidgetLoaded(final WidgetLoaded event)
+	{
+		final int groupId = event.getGroupId();
+
+		if (groupId != InterfaceID.TOA_RAID)
+		{
+			return;
+		}
+
+		Widget widget = client.getWidget(InterfaceID.TOA_RAID, CHILD_ID_KEPHRI);
+
+		if (widget != null)
+		{
+			kephriPathLevel = Integer.parseInt(widget.getText());
+		}
+
+		widget = client.getWidget(InterfaceID.TOA_RAID, CHILD_ID_AKKHA);
+
+		if (widget != null)
+		{
+			akkhaPathLevel = Integer.parseInt(widget.getText());
+		}
+
+		widget = client.getWidget(InterfaceID.TOA_RAID, CHILD_ID_BABA);
+
+		if (widget != null)
+		{
+			babaPathLevel = Integer.parseInt(widget.getText());
+		}
+
+		widget = client.getWidget(InterfaceID.TOA_RAID, CHILD_ID_ZEBAK);
+
+		if (widget != null)
+		{
+			zebakPathLevel = Integer.parseInt(widget.getText());
+		}
```

Comment:
It would be good to check the path level based on the player's current room when in a path as well, to support things like disconnects.

Reply:
Can you elaborate on how disconnecting affects the path level? I don't think I've disconnected before. Is it a party thing?

Reply:
I think you could miss the nexus room widget events under some really tight conditions of reconnecting in that room but being moved to next before loading.

Either way, it seems nicer to me to track it closer to when its used.